middleware/secure.go

```diff
@@ -108,7 +115,10 @@ func SecureWithConfig(config SecureConfig) echo.MiddlewareFunc {
 				res.Header().Set(echo.HeaderStrictTransportSecurity, fmt.Sprintf("max-age=%d%s", config.HSTSMaxAge, subdomains))
 			}
 			if config.ContentSecurityPolicy != "" {
-				res.Header().Set(echo.HeaderContentSecurityPolicy, config.ContentSecurityPolicy)
+				if !config.CSPReportOnly {
+					res.Header().Set(echo.HeaderContentSecurityPolicy, config.ContentSecurityPolicy)
+				}
+				res.Header().Set(echo.HeaderContentSecurityPolicyReportOnly, config.ContentSecurityPolicy)
```

Comment:
Without an else branch, you actually end up applying both headers. Is that the intent here? Doesn't sound like it from either the description or the comment above `CSPReportOnly`. Also from what I read at https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy-Report-Only it sounds like using them together would not produce the intended result.

Reply:
🤦‍♂️ will fix it shortly

Reply:
@alexaandru - please check

Reply:
Looks good now @kumarharsh ! Thanks! cc: @vishr @im-kulikov